cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,367 @@
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Approach
+
+The XML representation used is deliberately verbose in favor of readability
+and deterministic processing.
+
+Preservation of type information for CloudEvent attributes is supported allowing
+custom extensions to be communicated without type loss.
+
+A schema-less approach has been taken favoring convention over rigid document
+structure.
+
+The namespace `http://cloudevents.io/xmlformat/V1` MUST be used.
+
+When namespace prefixes are used a prefix of `ce` is preferred but MUST NOT
+be expected from an XML document processing perspective.
+
+An XML document preamble SHOULD be included to ensure deterministic processing.
+
+## 2. Attributes
+
+The CloudEvents type system is mapped to a set of CloudEvent specific
+type designators as follows :
+
+| CloudEvents Type  |  XML Format Type | XML Schema Type | Notes |
+| :-----------------| :----------------| :--------------- | :---- |
+| Boolean       | ce:boolean | [xs:boolean][xml-primitives] | `true` or `false` |
+| Integer       | ce:int | [xs:int][xml-primitives] | |
+| String        | ce:string | [xs:string][xml-primitives] | |
+| Binary        | ce:binary | [xs:base64Binary][xml-primitives] |  |
+| URI           | ce:uri | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986]| |
+| URI-reference | ce:uriRef | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986] | |
+| Timestamp     | ce:timestamp | [xs:dateTime][xml-primitives] following [RFC 3339][rfc3339] | |
+
+Each CloudEvent context attribute is represented as an XML element whose local
+name exactly matches that of the attribute.
+
+See the [envelope](#4-envelope) for special handing of the `specversion` context attribute.
+
+Extension attributes MUST be decorated with the appropriate CloudEvent format
+designators using using the `xsi:type` XML attribute, this allows them to be exchanged
+without loss of type information.
+
+REQUIRED and OPTIONAL context attributes SHOULD NOT be decorated with an `xsi:type`.
+
+No other XML element attributes are expected, if present they MUST be ignored during
+processing.
+
+``` xml
+    ...
+    <id>AAABBBCCCNNN0000</id>
+    ..
+    <time>2021-08-14T14:30:22-08:00</time>
+    ..
+    <myextension xsi:type="ce:string">my extension value</myextension>
+    <myboolean xsi:type="ce:boolean">false</myboolean>
+```
+
+## 3. Data
+
+The data portion of a CloudEvent follows a similar model to that employed by
+the [JSON Format specification][json-format]. A `data` element MUST be used to
+encapsulate the payload.
+
+An `xsi:type` is used to discrimate the payload type and MUST be present.
+
+The following data representations are supported:
+
+### 3.1 Binary Data
+
+Binary data MUST be carried in an element with an defined type of `ce:binary`.
```

Comment:
While the "types" table above says it needs to be base64 encoded, it might be good to remind people of this:
```
Note, the data MUST be base64 encoded per the ce:binary definition in the #attributes section. 
```

Reply:
on the fence as to the need for that.

Reply:
Just didn't want people to miss that link between the two spots in the spec.

Reply:
I an going to tweak this to be xs:base64Binary, the types of 'data' use the xs types; where as the types of attributes use the ce types.

Why i hear you ask ... well, i just (re)discovered that xs:string is not the same as a ce:string ;-)